yb-voyager/src/anon/sql_anonymizer.go

```diff
@@ -2233,65 +2151,119 @@ func (a *SqlAnonymizer) anonymizeColumnDefNode(cd *pg_query.ColumnDef) (err erro
-			if err != nil {
-				return fmt.Errorf("anon coldef default float value: %w", err)
-			}
-			// Replace the float value node with the string node for anonymized value
-			aConst.Val = &pg_query.A_Const_Sval{
-				Sval: &pg_query.String{Sval: hashedVal},
-			}
-			continue
+	// Handle Integer nodes (e.g., internallength = 1548)
+	if integer := arg.GetInteger(); integer != nil {
+		hashedVal, err := a.registry.GetHash(prefix, fmt.Sprintf("%d", integer.Ival))
+		if err != nil {
+			return fmt.Errorf("anon constant integer value: %w", err)
 		}
+		// Replace the integer node with a string node to preserve parse tree integrity
+		*arg = pg_query.Node{
+			Node: &pg_query.Node_String_{
+				String_: &pg_query.String{Sval: hashedVal},
+			},
+		}
+		return nil
+	}
 
-		// Handle boolean literals in default values
-		if boolval := aConst.GetBoolval(); boolval != nil {
-			// Get the boolean value as string and hash it
-			boolVal := boolval.String()
-			hashedVal, err := a.registry.GetHash(CONST_KIND_PREFIX, boolVal)
-			if err != nil {
-				return fmt.Errorf("anon coldef default boolean value: %w", err)
-			}
-			// Replace the boolean value node with the string node for anonymized value
-			aConst.Val = &pg_query.A_Const_Sval{
-				Sval: &pg_query.String{Sval: hashedVal},
-			}
-			continue
+	return nil
+}
+
+// anonymizeAConst is a shared helper function to anonymize A_Const nodes
+// This eliminates code duplication between literalNodesProcessor and anonymizeConstantValue
+func (a *SqlAnonymizer) anonymizeAConst(ac *pg_query.A_Const, prefix string) error {
+	if ac == nil || ac.Val == nil {
+		return nil
+	}
+
+	// Handle string literals
+	if sval := ac.GetSval(); sval != nil {
+		hashedVal, err := a.registry.GetHash(prefix, sval.Sval)
+		if err != nil {
+			return fmt.Errorf("anon A_Const string: %w", err)
+		}
+		sval.Sval = hashedVal
+		return nil
+	}
+
+	// Handle integer literals
+	if ival := ac.GetIval(); ival != nil {
+		intVal := fmt.Sprintf("%d", ival.Ival)
+		hashedVal, err := a.registry.GetHash(prefix, intVal)
+		if err != nil {
+			return fmt.Errorf("anon A_Const int: %w", err)
+		}
+		// Replace with string value
+		ac.Val = &pg_query.A_Const_Sval{
+			Sval: &pg_query.String{Sval: hashedVal},
+		}
+		return nil
+	}
+
+	// Handle float literals
+	if fval := ac.GetFval(); fval != nil {
+		hashedVal, err := a.registry.GetHash(prefix, fval.Fval)
+		if err != nil {
+			return fmt.Errorf("anon A_Const float: %w", err)
+		}
+		// Replace with string value
+		ac.Val = &pg_query.A_Const_Sval{
+			Sval: &pg_query.String{Sval: hashedVal},
+		}
+		return nil
+	}
+
+	// Handle boolean literals
+	if boolval := ac.GetBoolval(); boolval != nil {
+		boolVal := boolval.String()
+		hashedVal, err := a.registry.GetHash(prefix, boolVal)
+		if err != nil {
+			return fmt.Errorf("anon A_Const bool: %w", err)
 		}
+		// Replace with string value
+		ac.Val = &pg_query.A_Const_Sval{
+			Sval: &pg_query.String{Sval: hashedVal},
+		}
+		return nil
```

Comment:
Should we have different prefixes for different kind of values, like `const_string_`, `const_int_`, etc.. to have distinction of all types of values